cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -391,59 +435,29 @@ public TimeSeries getTimeseries(String page, int pageSize, String names, String
 
             if (pageSize > 0) {
                 query.limit(DSL.val(pageSize + 1));
+                query2.limit(DSL.val(pageSize + 1));
             }
 
-            SelectConditionStep<Record3<Timestamp, Double, BigDecimal>> finalQuery = query;
-            logger.fine(() -> finalQuery.getSQL(ParamType.INLINED));
-
             if (includeEntryDate) {
-                SelectConditionStep<Record4<Timestamp, Double, BigDecimal, Timestamp>> query2 = dsl.select(
-                                    dateTimeCol,
-                                    valueCol,
-                                    qualityNormCol,
-                                    dataEntryDate
-                            )
-                            .from(AV_TSV_DQU.AV_TSV_DQU)
-                            .where(dateTimeCol
-                                    .greaterOrEqual(CWMS_UTIL_PACKAGE.call_TO_TIMESTAMP__2(
-                                            DSL.nvl(DSL.val(tsCursor == null ? null :
-                                                            tsCursor.toInstant().toEpochMilli()),
-                                                    DSL.val(beginTime.toInstant().toEpochMilli())))))
-                            .and(dateTimeCol
-                                    .lessOrEqual(CWMS_UTIL_PACKAGE.call_TO_TIMESTAMP__2(
-                                            DSL.val(endTime.toInstant().toEpochMilli())))
-                            .and(AV_TSV_DQU.AV_TSV_DQU.CWMS_TS_ID.equalIgnoreCase(names))
-                            .and(AV_TSV_DQU.AV_TSV_DQU.OFFICE_ID.eq(office))
-                            .and(AV_TSV_DQU.AV_TSV_DQU.UNIT_ID.equalIgnoreCase(unit))
-                            .and(AV_TSV_DQU.AV_TSV_DQU.VERSION_DATE.eq(versionDate == null ? null :
-                                    Timestamp.from(versionDate.toInstant())))
-                                );
-
-                if (pageSize > 0) {
-                    query2.limit(DSL.val(pageSize + 1));
-                }
-                query2.forEach(tsRecord -> {
-                    assert timeseries != null;
-                    timeseries.addValue(
-                        tsRecord.getValue(dateTimeCol),
-                        tsRecord.getValue(valueCol),
-                        tsRecord.getValue(qualityNormCol).intValue(),
-                        tsRecord.getValue(dataEntryDate)
-                    );
-                });
+                logger.fine(() -> query2.getSQL(ParamType.INLINED));
+                final TimeSeriesWithDate timeSeries = new TimeSeriesWithDate(timeseries);
+                query2.forEach(tsRecord -> timeSeries.addValue(
```

Comment:
we don't need to solve this now, but definitely before we add requesting any text notes as well. There has to be a better way to handle this with the builders.